acquia_lift.admin.wizard.inc

```diff
@@ -243,10 +241,68 @@ function acquia_lift_personalize_campaign_wizard_variations_alter(&$form, &$form
  * Alter hook for the targeting portion of the campaign wizard.
  */
 function acquia_lift_personalize_campaign_wizard_targeting_alter(&$form, &$form_state, $form_id) {
-  if (!empty($form['visitor_context']) && $form['visitor_context']['#type'] == 'select') {
+  if (!empty($form['targeting']['visitor_context']) && $form['targeting']['visitor_context']['#type'] == 'select') {
     module_load_include('inc', 'acquia_lift', 'acquia_lift.ui');
-    acquia_lift_chosenify_element($form['visitor_context']);
+    acquia_lift_chosenify_element($form['targeting']['visitor_context']);
+  }
+
+  // Acquia Lift Target agents allow for audience designation to be used across
+  // the agent.
+  $agent_data = $form['#agent'];
+  if ($agent_data->plugin !== 'acquia_lift_target') {
+    return;
   }
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.admin');
+  $option_set = acquia_lift_get_option_set_for_targeting($agent_data->machine_name);
+
+  if (empty($option_set)) {
+    drupal_set_message(t('You must <a href="!variations_url">create variation sets</a> before targeting.', array(
+      '!variations_url' => url('admin/structure/personalize/manage/' . $agent_data->machine_name . '/variations'),
+    )), 'warning');
+    return;
+  }
+  // Take over the form validation and submit handling.
+  $form['#validate'] = array('acquia_lift_personalize_campaign_wizard_targeting_validate');
+  $form['#submit'] = array('acquia_lift_personalize_campaign_wizard_submit');
+
+  unset($form['targeting']['option_sets']);
+  $form['audiences'] = array(
+    '#type' => 'container',
+    '#tree' => TRUE,
+  );
+  // Load all audience rules for the agent.
+  if (isset($option_set->targeting)) {
+    foreach ($option_set->targeting as $rule_id => $rule) {
+      $form['audiences'][$rule_id] = array(
+        '#type' => 'container',
+        '#theme' => 'acquia_lift_card',
+        '#title' => isset($rule['label']) ? $rule['label'] : 'Audience: ' . $rule_id,
+        '#collapsible' => TRUE,
+        '#collapsed' => TRUE,
+      );
+      $form['audiences'][$rule_id]['header'] = array(
+        '#type' => 'container'
+      );
+      $form['audiences'][$rule_id]['header']['delete'] = array(
+        '#markup' => l(t('Delete'), 'admin/structure/personalize/manage/' . $agent_data->machine_name . '/audience/' . $rule_id . '/delete', array(
+          'attributes' => array(
+            'class' => array('acquia-lift-delete')
+          ),
+        )),
+      );
+      _acquia_lift_personalize_campaign_wizard_targeting_audience($form['audiences'][$rule_id]['details'], $form_state, $agent_data, $rule_id, $rule);
```

Comment:
Could we change this to 
`$form['audiences'][$rule_id]['details'] = _acquia_lift_personalize_campaign_wizard_targeting_audience($form_state, ...)`
That function already returns the form array and it seems confusing to pass the section of the form by reference here.